src/selectors/entities/channel_categories.ts

```diff
@@ -248,7 +248,12 @@ export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channe
                 remaining = limitPref;
             }
 
+            const currentChannel = filtered.find((channel) => channel.id === currentChannelId);
             const slicedChannels = filtered.slice(0, remaining);
+            if (currentChannel && !slicedChannels.includes(currentChannel)) {
+                slicedChannels.pop();
```

Comment:
Which channel gets popped off by this? Maybe we should just let the count be one over in this case?